templates/hardhat/solidity/scripts/interact.ts

```diff
@@ -0,0 +1,37 @@
+// Script that interacts with a Greeter contract
```

Comment:
```suggestion
// Script that interacts with a Greeter contract
import { ethers, network } from 'hardhat';
```

add this and you can remove all `hre` instances & the TS analysis will work. Same for other script files